### PR TITLE
Improve benchmark experience

### PR DESCRIPTION
**Pull request by StarpTech**
- Support running multiple scripts on windows and linux via concurrently package.
- Use https://github.com/StarpTech/branch-comparer to run benchmarks in different branches

example

__*the frames of the gif were reduced because of file size.__

### CODE REVIEW

**Comment by StarpTech** [comment]
@mcollina I had to remove the previous branch because I had some trouble to resolve a merge. That's the final PR.

**Comment by allevo** [comment]
Wow! nice!

Could we run the bench multiple times and keep last result?

**Comment by StarpTech** [comment]
You can run the command just two times in a row.

**Comment by StarpTech** [comment]
But should be easy to adopt it.

**Comment by StarpTech** [comment]
@allevo has now support for it. Run:

```sh
branchcmp --rounds 2 // cli output
branchcmp --file --rounds 2 // file output
```

**Comment by StarpTech** [comment]
@mcollina @delvedor any chance to merge this ?

File: package.json [hunk 1]
```diff
@@ -14,7 +14,7 @@
     "test": "npm run lint && npm run unit && npm run typescript",
     "coverage": "npm run unit -- --cov --coverage-report=html",
     "coveralls": "npm run unit --  --cov",
-    "benchmark": "node ./examples/$npm_config_usingfile & pid=$! && autocannon -c 100 -d 5 -p 10 localhost:3000/ && kill -9 $pid"
+    "benchmark": "node ./node_modules/concurrently -k -s first \"node ./examples/simple.js\" \"node ./node_modules/autocannon -c 100 -d 5 -p 10 localhost:3000/\""
```
[the remark this answers is not in the document]
Comment (StarpTech) [reply package.json]: I'm using npx shipped with 5.2 to run local binaries

File: docs/Benchmarking.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+### Simple
+```sh
+npm run benchmark
+```
+
+### Compare multiple branches
+```sh
+branchcmp --rounds 2
+```
+_Answer the second question with `npm run benchmark`._
```
[the remark this answers is not in the document]
Comment (StarpTech) [reply docs/Benchmarking.md]: Devs are lazy :D

File: package.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (jsumners) [reply package.json]: npm automatically adds module "binaries" to the `PATH` when running them via npm scripts. Thus, the following two snippets are equivalent:

```json
{
  "scripts:" {
    "lint": "standard"
  }
}
```

```json
{
  "scripts:" {
    "lint": "node ./node_modules/.bin/standard"
  }
}
```

[the remark this answers is not in the document]
Comment (StarpTech) [reply package.json]: Nice, wasn't aware of that.

[the remark this answers is not in the document]
Comment (StarpTech) [reply package.json]: But it seems not to work with `npx` to test multiple nodejs versions

```
 "benchmark": "concurrently -k -s first \"node ./examples/simple.js\" \"autocannon -c 100 -d 5 -p 10 localhost:3000/\""
```
`npx -p node@9 -- npm run benchmark`

**Will Fail** but with `npx` it works I think the reason is that `npx` will guarantee that the module is installed locally and doesn't point to globals. So I will stay with `npx`

File: docs/Benchmarking.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+The modules we'll use:
+- [Autocannon](https://github.com/mcollina/autocannon): A HTTP/1.1 benchmarking tool written in node.
+- [Branch-comparer](https://github.com/StarpTech/branch-comparer): Checkout multiple git branches, execute scripts and log the results.
+- [Concurrently](https://github.com/kimmobrunfeldt/concurrently): Run commands concurrently.
+
+## Installation
+
+```sh
+npm i -g branch-comparer
```
[the remark this answers is not in the document]
Comment (StarpTech) [reply docs/Benchmarking.md]: Done

File: docs/Benchmarking.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (StarpTech) [reply docs/Benchmarking.md]: Fixed `branchcmp` will restore to the specific sh1 commit

**Comment by StarpTech** [comment]
@mcollina Reminder - I updated all places.

File: docs/Benchmarking.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (StarpTech) [reply docs/Benchmarking.md]: I added support for autocomplete and command history. If you have any ideas how we can simplify it open an issue.

**Comment by StarpTech** [comment]
Added support for benchmarking of different node.js versions via `npx`

File: docs/Benchmarking.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (StarpTech) [reply docs/Benchmarking.md]: You can pass the script as `-s` flag

**Comment by mcollina** [comment]
@StarpTech this is very very good. However it would be fantastic to have a `npm run bench-against-master` command that benchmark whatever is here vs master. This would be very easy to tell people to do when they submit a PR.

The UI of the tool itself becomes very complex for me to use, as I have 48 different branches on my git repo: finding the ones to check takes a lot of time. Not sure what can be done there, maybe some form of tab-completion to filter?

**Comment by StarpTech** [comment]
@mcollina yes, that's correct but that's easy to fix we can implement a flag which compares always the master with your current branch (Github flow) but let's merge this solution for the rest of the people I will update branchcmp later today.

**Comment by StarpTech** [comment]
@mcollina that's your command `branchcmp --rounds 2 --gitflow --script "npm run benchmark"` or
`branchcmp --r 2 -g -s "npm run benchmark"`

**Comment by mcollina** [comment]
can you send a PR adding branchcmp and your command to package.json? Lazy devs.

**Pull request merged**